### DISCUSSION TITLE
can bun use hard links to manage dependencies like pnpm does?

### DISCUSSION THREAD

**Discussion by haosmos** [oven-sh/bun#4999] [General]
pnpm uses hard links, which allows you to reuse installed packages in different projects, thereby saving a huge amount of disk space. Can bun do the same? If not, do you plan to implement a similar mechanism for the dependency manager?

**Comment by Bhanusingh99** [comment]
Subject: Collaboration Opportunity: Promote Your Bun Products to Our Engaged Tech Community!

Dear [Bun Corporation's Name],

I hope this message finds you well. My name is Bhanu Singh, and I represent a dynamic duo of tech enthusiasts and micro Twitter influencers. We are excited to reach out to Bun Corporation with a unique collaboration proposal that we believe can benefit both of our brands.

Firstly, we wanted to applaud your incredible Bun products, which we've personally found to be incredibly useful and innovative. Our tech-savvy audience on Twitter is always on the lookout for practical solutions that can improve their daily lives, and we believe Bun products align perfectly with their needs.

Here's a brief overview of what we can offer:

1. **Engaged Twitter Community:** As micro-influencers in the tech niche, we have cultivated a loyal and active following within the Twitter tech community. On average, our tweets garner approximately 30,000 views per post, and this number is steadily growing.

2. **Specialized Content Creation:** We are eager to create two engaging and informative posts specifically dedicated to Bun products. These posts will highlight the unique features and benefits of using your products, ensuring they reach a highly receptive audience.

3. **Impressive Reach:** Our most recent tweet about bun-related content garnered an astonishing 15,000 views within just 6 hours, showcasing our ability to capture our audience's attention effectively.

4. **Affordable Pricing:** We understand the value of effective marketing and would like to offer our services at a competitive rate of $250 for both posts.

We believe that by partnering with us, Bun Corporation can tap into a highly targeted and enthusiastic customer base, while we can continue to provide our followers with valuable and innovative tech solutions.

If you're interested in discussing this collaboration further or have any questions, please feel free to reach out to us at [Your Email Address]. We would be delighted to explore how we can work together to elevate the visibility of Bun products in the tech community.

Thank you for considering our proposal, and we look forward to the possibility of collaborating with Bun Corporation.

Warm regards,

[Bhanu singh]
[Your Twitter Handles
@Bhanu_here
@maybeshalinii
]
[bhanusngh12345@gmail.com]

**Reply by birkskyum** [reply]
Please stop spamming this 
- https://github.com/oven-sh/bun/discussions/4313#discussioncomment-6972012

**Comment by mehradotdev** [comment]
hi, any updates on this?

**Comment by dylan-conway** [comment]
In Bun v1.2.19 we shipped support for `--linker=isolated` in `bun install`. Isolated installs are like pnpm installs: packages are hardlinked to a store in `./node_modules/.bun` and dependencies are symlinked to the packages in the store. Note: on macOS we default to clonefile instead of hardlinks so you'll also need to pass `--backend=hardlink` to `bun install` if you want to ensure hardlinks are used on all platforms.

To enable isolated installs use `--linker=isolated` or add the following to `bunfig.toml`. In Bun v1.3 we will likely make isolated installs the default.

```toml
[install]
linker = "isolated"
```
